cardano-crypto-tests/src/Test/Crypto/VRF.hs

```diff
@@ -59,8 +69,183 @@ tests =
+          -- pi = 926e895d308f5e328e7aa159c06eddbe56d06846abf5d98c2512235eaa57fdce35b46edfc655bc828d44ad09d1150f31374e7ef73027e14760d42e77341fe05467bb286cc2c9d7fde29120a0b2320d04
+          testCase "generated golden test vector: vrf_ver13_standard_12" $
+            checkVer13TestVector "vrf_ver13_standard_12"
+        ]
     ]
 
+checkVer03TestVector :: FilePath -> Assertion
+checkVer03TestVector file = do
+  dir <- getCurrentDirectory
+  str <- readFile $ dir <> "/test_vectors/" <> file
+  let testVectorE = Read.readMaybe @VRFTestVector str
+  assertBool ("parsing test vector: " <> file <> " not successful") $ isJust testVectorE
+  let VRFTestVector {..} = fromJust testVectorE
+  signKey <- Ver03.skFromBytes signingKey
+  verKey <- Ver03.vkFromBytes verifyingKey
+  let bytesEq left right = case left of
+        Just left' ->
+          Ver03.outputBytes left' @?= Ver03.outputBytes right
+        Nothing ->
+          assertBool ("left side in byteEq gave Nothing") False
+
+  name @?= algorithmNameVRF (Proxy :: Proxy PraosVRF)
+  version @?= "ietfdraft03"
+  ciphersuite @?= "ECVRF-ED25519-SHA512-Elligator2"
+  proof' <- Ver03.proofFromBytes proof
+  let hash' = Ver03.outputFromBytes hash
+  -- prove signKey msg -> proof
+  Ver03.prove signKey message @?= Just proof'
+  -- signKey -> verKey
+  Ver03.skToVerKey signKey @?= verKey
+  -- proof -> hashed msg
+  bytesEq (Ver03.outputFromProof proof') hash'
+  -- verify verKey proof msg -> hashed msg
+  bytesEq (Ver03.verify verKey proof' message) hash'
+
+checkVer13TestVector :: FilePath -> Assertion
+checkVer13TestVector file = do
+  dir <- getCurrentDirectory
+  str <- readFile $ dir <> "/test_vectors/" <> file
+  let testVectorE = Read.readMaybe @VRFTestVector str
+  assertBool ("parsing test vector: " <> file <> " not successful") $ isJust testVectorE
+  let VRFTestVector {..} = fromJust testVectorE
+  let signKey = Ver13.skFromBytes signingKey
+  let verKey = Ver13.vkFromBytes verifyingKey
+  let bytesEq left right = case left of
+        Just left' ->
+          Ver13.outputBytes left' @?= Ver13.outputBytes right
+        Nothing ->
+          assertBool ("left side in byteEq gave Nothing") False
+
+  name @?= algorithmNameVRF (Proxy :: Proxy PraosBatchCompatVRF)
+  version @?= "ietfdraft13"
+  ciphersuite @?= "ECVRF-ED25519-SHA512-Elligator2"
+  -- prove signKey msg -> proof
+  let proof' = Ver13.proofFromBytes proof
+  let hash' = Ver13.outputFromBytes hash
+  Ver13.prove signKey message @?= Just proof'
+  -- signKey -> verKey
+  Ver13.skToVerKey signKey @?= verKey
+  -- proof -> hashed msg
+  bytesEq (Ver13.outputFromProof proof') hash'
+  -- verify verKey proof msg -> hashed msg
+  bytesEq (Ver13.verify verKey proof' message) hash'
+
+data VRFTestVector = VRFTestVector
+  { name :: String
+  , version :: String
+  , ciphersuite :: String
+  , signingKey :: BS.ByteString
+  , verifyingKey :: BS.ByteString
+  , message :: BS.ByteString
+  , proof :: BS.ByteString
+  , hash :: BS.ByteString
```

Comment:
I'd like to ask you to use camel case for `cipherSuite` and to use a common prefix for the fields:
```suggestion
  { testVectorName :: String
  , testVectorVersion :: String
  , testVectorCipherSuite :: String
  , testVectorSigningKey :: BS.ByteString
  , testVectorVerifyingKey :: BS.ByteString
  , testVectorMessage :: BS.ByteString
  , testVectorProof :: BS.ByteString
  , testVectorHash :: BS.ByteString
```

If `testVector` is too long of a prefix for your liking, I'd be ok with an abbreviation prefix of `tv`.

Without such prefix it is unclear where the binding is coming from (especially with RecordWildCards) and it can easily shadow binding with such a common name as `name` or `version` or `hash`. We want to avoid that

Reply:
valid points. done!